app/lib/shared/tool_env.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:pana/pana.dart';
+
+import 'configuration.dart';
+
+const _maxCount = 100;
+
+ToolEnvRef _current;
+
+class ToolEnvRef {
```

Comment:
Add some documentation.

This class seems like it would be easy to write a test case for.